opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -581,4 +591,23 @@ private void requestUserPassword() {
     public void onPasswordVerified() {
         startSprayForm(selectedFeature);
     }
+
+    @Override
+    public void waitForUserLocation() {
+        Location location = listTaskView.getUserCurrentLocation();
+        Log.d(TAG, "user location: " + location);
+        if (location == null) {
+            listTaskView.showProgressDialog(R.string.narrowing_location_title, R.string.narrowing_location_message);
+            new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    waitForUserLocation();
```

Comment:
Is it deterministic that at some point the location will not be null?

If not, this function could recurse indefinitely.